internal/namespaces/container/v1beta1/custom_deploy.go

```diff
@@ -0,0 +1,307 @@
+package container
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	dockerTypes "github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/scaleway/scaleway-cli/v2/internal/core"
+	"github.com/scaleway/scaleway-cli/v2/internal/tasks"
+	container "github.com/scaleway/scaleway-sdk-go/api/container/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+type containerDeployRequest struct {
+	Region scw.Region
+
+	Name        string
+	Dockerfile  string
+	BuildSource string
+	Port        uint32
+}
+
+func containerDeployCommand() *core.Command {
+	return &core.Command{
+		Short:     `Deploy a container`,
+		Long:      `Automatically build and deploy a container.`,
+		Namespace: "container",
+		Resource:  "deploy",
+		Groups:    []string{"workflow"},
+		ArgsType:  reflect.TypeOf(containerDeployRequest{}),
+		ArgSpecs: core.ArgSpecs{
+			{
+				Name:  "name",
+				Short: "Name of the application",
+				Default: func(ctx context.Context) (value string, doc string) {
+					currentDirection, err := os.Getwd()
+					if err != nil {
+						return "", ""
+					}
+
+					name := filepath.Base(currentDirection)
+					if name == "." {
+						return "", ""
+					}
+
+					name = "app-" + name
+					return name, name
+				},
+			},
+			{
+				Name:    "dockerfile",
```

Comment:
Do we plan to test with buildpack in the future too? Is this design compatible with buildpack in case we have no dockerfile in the code.

Reply:
This PR is a MVP of the feature, buildpack will come after